### ISSUE TITLE
Serdes v2 is not transferring `report_card.result_metadata` on models

### ISSUE THREAD

**Issue by luizarakaki** [metabase/metabase#26150]
**Describe the bug**
Models rely on `report_card.result_metadata` to store metadata. Currently, serialization is not transferring this field, so users lose the customized metadata when loading models.

**To Reproduce**
1. Go to a source instance, create a model and add metadata (change field display name, field description, visibility type or semantic type)
2. Serialize and deserialize the model
3. Open the model in the new instance. The metadata isn't there.

**Expected behavior**
It should transfer result_metadata (at least for models), since this field supports models.

**Severity**
P2

**Comment by flamber** [comment]
Basically a duplicate of #20282

**Comment by luizarakaki** [comment]
This is also creating problems with nested queries that depends on result_metadata (like binning):

```
2022-10-28 10:57:32,884 ERROR middleware.catch-exceptions :: Error processing query: Cannot update binned field: query is missing source-metadata
{:database_id 1,
 :started_at #t "2022-10-28T10:57:32.561749-03:00[America/Sao_Paulo]",
 :via
 [{:status :failed,
   :class clojure.lang.ExceptionInfo,
   :error "Cannot update binned field: query is missing source-metadata"
```

**Comment by luizarakaki** [comment]
> Basically a duplicate of #20282

Yeah, but let's keep two separate issues. We need to solve this for v2 while v1 won't be fixed.

**Comment by luizarakaki** [comment]
Closed with #26154
